src/ModbusClientTCPasync.cpp

```diff
@@ -333,7 +312,7 @@ void ModbusClientTCPasync::handleSendingQueue() {
     RequestEntry* r = *i;
     if (send(r)) {
       // after sending, update timeout value, add to other queue and remove from this queue
-      r->timeout = millis();
+      r->sentTime = millis();  //@Miq1 sentTime is already set, changing here doesn't work?
```

Comment:
This apparently doesn't work on my side. `sentTime` stays zero (if not set with the added code above) and timeouts.

Reply:
Completely your code, I did not change a bit here?

Reply:
I know. But somehow it doesn't work!

I had to set sentTime (changed variable name to more appropriate) at enqueuing instead of at sending because at sending it doesn't stick.

Reply:
I will have a look at this tomorrow - I am already in bed 😄

Reply:
So in this block
```cpp
void ModbusClientTCPasync::handleSendingQueue() {
  // ATTENTION: This method does not have a lock guard.
  // Calling sites must assure shared resources are protected
  // by mutex.

  // try to send everything we have waiting
  std::list<RequestEntry*>::iterator i = txQueue.begin();
  while (i != txQueue.end()) {
    // get the actual element
    RequestEntry* r = *i;
    if (send(r)) {
      // after sending, update timeout value, add to other queue and remove from this queue
      r->sentTime = millis();  //@Miq1 sentTime is already set, changing here doesn't work?
      rxQueue[r->head.transactionID] = r;      // push request to other queue
      i = txQueue.erase(i);  // remove from toSend queue and point i to next request
    } else {
      // sending didn't succeed, try next request
      ++i;
    }
  }
}
```
the `sentTime` will be 0 and the request will timeout.

Reply:
I think I know why: the txQueue holds elements of type ``RequestEntry*``. ``r`` is the same, but you do the assignment ``r = *i;``. This should have triggered a compiler warning, though.